extensions/podman/packages/api/src/podman-extension-api.d.ts

```diff
@@ -0,0 +1,23 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { ProviderContainerConnection, RunOptions, RunResult } from '@podman-desktop/api';
+
+export interface PodmanExtensionApi {
+  exec(args: string[], connection?: ProviderContainerConnection, options?: RunOptions): Promise<RunResult>;
+}
```

Comment:
```suggestion
export interface PodmanRunOptions extends RunOptions {
  connection?: ProviderContainerConnection;
}

export interface PodmanExtensionApi {
  exec(args: string[], options?: PodmanRunOptions): Promise<RunResult>;
}
```

I must admit I would have a preference for having everything in the options (personal preference). cc @benoitf

Reply:
I agree, a single option parameter is better, even more when it's optional